tests/python/test_indices.py

```diff
@@ -19,3 +19,22 @@ def test_indices():
     assert mapping_b == {0: 1, 1: 0}
     # Note that b is column-major:
     # the virtual first index exposed to the user comes second in memory layout.
+
+
+@ti.host_arch_only
+def test_float_as_index():
+    a = ti.var(ti.f32, (8, 5))
+
+    @ti.kernel
+    def func():
+        i = 6.66
+        j = 3
+        I = ti.Vector([2, 1])
+        for _ in range(1):  # prevent constant fold
+            a[i, j] = 233
+            a[I + ti.Vector([1, 3.0])] = 666
+
+    func()
+
+    assert a[6, 3] == 233
```

Comment:
Is it guaranteed that `int32(6.66) == 6` on all platforms?

Reply:
Yes, that's how a Python `int` does, and I believe backends should follow that too.